notebook/agentchat_mongodb_RetrieveChat.ipynb

```diff
@@ -0,0 +1,538 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Using RetrieveChat Powered by MongoDB Atlas for Retrieve Augmented Code Generation and Question Answering\n",
+    "\n",
+    "AutoGen offers conversable agents powered by LLM, tool or human, which can be used to perform tasks collectively via automated chat. This framework allows tool use and human participation through multi-agent conversation.\n",
+    "Please find documentation about this feature [here](https://microsoft.github.io/autogen/docs/Use-Cases/agent_chat).\n",
+    "\n",
+    "RetrieveChat is a conversational system for retrieval-augmented code generation and question answering. In this notebook, we demonstrate how to utilize RetrieveChat to generate code and answer questions based on customized documentations that are not present in the LLM's training dataset. RetrieveChat uses the `RetrieveAssistantAgent` and `RetrieveUserProxyAgent`, which is similar to the usage of `AssistantAgent` and `UserProxyAgent` in other notebooks (e.g., [Automated Task Solving with Code Generation, Execution & Debugging](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_auto_feedback_from_code_execution.ipynb)). Essentially, `RetrieveAssistantAgent` and  `RetrieveUserProxyAgent` implement a different auto-reply mechanism corresponding to the RetrieveChat prompts.\n",
+    "\n",
+    "## Table of Contents\n",
+    "We'll demonstrate six examples of using RetrieveChat for code generation and question answering:\n",
+    "\n",
+    "- [Example 1: Generate code based off docstrings w/o human feedback](#example-1)\n",
+    "\n",
+    "````{=mdx}\n",
+    ":::info Requirements\n",
+    "Some extra dependencies are needed for this notebook, which can be installed via pip:\n",
+    "\n",
+    "```bash\n",
+    "pip install pyautogen[retrievechat-mongodb] flaml[automl]\n",
+    "```\n",
+    "\n",
+    "For more information, please refer to the [installation guide](/docs/installation/).\n",
+    ":::\n",
+    "````\n",
+    "\n",
+    "Ensure you have a MongoDB Atlas instance.\n",
+    "\n",
+    "If not, a test version can quickly be deployed using Docker.\n",
+    "\n",
+    "`docker-compose.yml`\n",
+    "\n",
+    "```yml\n",
+    "version: '3.9'\n",
+    "\n",
+    "services:\n",
+    "  mongodb:\n",
+    "    image: mongodb/mongodb-atlas-local:latest\n",
+    "    restart: unless-stopped\n",
+    "    ports:\n",
+    "      - \"27017:27017\"\n",
+    "    environment:\n",
+    "      MONGODB_INITDB_ROOT_USERNAME: mongodb_user\n",
+    "      MONGODB_INITDB_ROOT_PASSWORD: mongodb_password\n",
```

Comment:
With this setting, I always get empty returns like below:
```
2024-06-30 21:18:02,516 - autogen.agentchat.contrib.vectordb.mongodb - INFO - Search index vector_index created successfully.
VectorDB returns doc_ids:  [[]]
No more context, will terminate.
```
Are you sure the free version of mongodb works?